components/raftstore-v2/src/operation/command/write/mod.rs

```diff
@@ -118,12 +118,14 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
 
     pub fn propose_pending_writes<T>(&mut self, ctx: &mut StoreContext<EK, ER, T>) {
         if let Some(encoder) = self.simple_write_encoder_mut().take() {
+            let header = encoder.header();
+            let validate_res = self.validate_command(header, None, &mut ctx.raft_metrics);
```

Comment:
validate_command is already called before.  But here what really matters is the compare_region_epoch?

Reply:
The matter here is that the peer should be leader now. I think call validate_command is safer and innocent.